File: docs/performance/fib3/fib.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int fib(int n) {
    if (n < 2)
        return n;
    return (fib(n - 1) + fib(n - 2));
}

int main(int argc, char **argv) {
    clock_t t0;
    int n = atoi(argv[1]);
    t0 = clock();
    int res = fib(n);
    double dt = ((double)(clock() - t0)) / CLOCKS_PER_SEC;
    printf("fib(%d) = %d (run time %0.2f seconds)\n", n, res, dt);
}

File: docs/performance/fib2/fib.py
import sys
import time


def fib(n):
    """Print the Fibonacci series up to n."""
    if n < 2:
        return n
    return fib(n-1) + fib(n-2)


if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: python fib.py n")
        sys.exit(0)
    n = int(sys.argv[1])
    t0 = time.time()
    res = fib(n)
    dt = time.time() - t0
    print(f"fib({n}) = {res} (run time {dt} seconds)")
